### PR TITLE
dreambooth upscaling fix added latents

### PR DESCRIPTION
**Pull request by williamberman**
re: https://github.com/huggingface/diffusers/issues/3658

Also added some docs re: recent much more successful run doing full fine tuning for stage II with larger batch sizes.
https://wandb.ai/williamberman/dreambooth/runs/4bwrpro7/overview?workspace=user-williamberman 

I didn't have as much success with LoRA for larger batch sizes but the docs note we recommend full finetuning stage II for faces anyway
https://wandb.ai/williamberman/dreambooth-lora/runs/cdtpa4l7?workspace=user-williamberman 

Unfortunately the docs are a bit "here's all the information" rather than "here's how to do a successful training run". But I think that's the best they can be right now

### CODE REVIEW

File: examples/dreambooth/train_dreambooth.py [hunk 1]
```diff
@@ -1212,14 +1211,8 @@ def compute_text_embeddings(prompt):
...
-                if unet.config.in_channels > channels:
-                    needed_additional_channels = unet.config.in_channels - channels
-                    additional_latents = randn_tensor(
-                        (bsz, needed_additional_channels, height, width),
-                        device=noisy_model_input.device,
-                        dtype=noisy_model_input.dtype,
-                    )
-                    noisy_model_input = torch.cat([additional_latents, noisy_model_input], dim=1)
+                if unet.config.in_channels == channels * 2:
+                    noisy_model_input = torch.cat([noisy_model_input, noisy_model_input], dim=1)
```
Comment (williamberman) [inline examples/dreambooth/train_dreambooth.py:1215]: Just doubling the existing latents is sufficient as the we use the same timestep for _both_ of the conditioning inputs! 

Theoretically, we could do something along the lines of training on two separate noising amounts for each set of channels or bias one towards a lower amount of noise as the pipeline's default is for a quarter of total noise to be added to the upsampled input.

However, this is the simplest implementation for now so I think it's ok!

File: docs/source/en/training/dreambooth.mdx [hunk 2]
```diff
@@ -540,10 +540,13 @@ upscaler to remove the new token from the instance prompt. I.e. if your stage I
 For finegrained detail like faces that aren't present in the original training set, we find that full finetuning of the stage II upscaler is better than 
 LoRA finetuning stage II.
 
-For finegrained detail like faces, we find that lower learning rates work best.
+For finegrained detail like faces, we find that lower learning rates along with larger batch sizes work best.
 
 For stage II, we find that lower learning rates are also needed.
 
+We found experimentally that the DDPM scheduler with the default larger number of denoising steps to sometimes work better than the DPM Solver scheduler
```
Comment (patrickvonplaten) [inline docs/source/en/training/dreambooth.mdx:547]: Nice!

File: docs/source/en/training/dreambooth.mdx [hunk 3]
```diff
@@ -682,8 +690,8 @@ accelerate launch train_dreambooth.py \
   --instance_prompt="a sks dog" \
   --resolution=256 \
   --train_batch_size=2 \
-  --gradient_accumulation_steps=2 \
-  --learning_rate=1e-8 \
+  --gradient_accumulation_steps=6 \
```
Comment (patrickvonplaten) [inline docs/source/en/training/dreambooth.mdx:693]: Cool, makes sense that bigger batch size will help here

**Review by patrickvonplaten — APPROVED** [review]
Very nice find!

**Review by pcuenca — APPROVED** [review]
Great experimentation!

**Pull request merged**